agent/util-scripts/pbench-make-result-tb

```diff
@@ -29,50 +35,57 @@ prefix=""
 xz_single_threaded=0
 eval set -- "${opts}"
 while true; do
-    case "${1}" in
+    opt="${1}"
+    shift;
+    case "${opt}" in
 	--result-dir)
-	    shift;
             if [[ -n "${1}" ]]; then
                 result_dir="${1}"
                 shift;
+            else
+            	missing_arg "${opt}"
             fi
             ;;
 	--target-dir)
-	    shift;
             if [[ -n "${1}" ]]; then
                 target_dir="${1}"
                 shift;
+            else
+            	missing_arg "${opt}"
             fi
             ;;
 	--user)
-	    shift;
             if [[ -n "${1}" ]]; then
                 user="${1}"
-                shift;
             fi
+            shift;
             ;;
         --prefix)
-            shift;
             if [[ -n "${1}" ]]; then
                 prefix="${1}"
-                shift;
             fi
+            shift;
```

Comment:
I don't understand: why are the first two treated differently from the next two? It seems to me that the treatment should be the same in all four cases.

Reply:
Because `--user` and `--prefix` are optional values provided by our caller, `pbench-move-results`, where-as the other parameters are required values for operation.